File: SuperheroApp/DataSource/CharactersDataSource.swift
//
//  CharactersDataSource.swift
//  SuperheroApp
//
//  Created by Balazs Varga on 2018. 02. 11..
//  Copyright © 2018. W.UP. All rights reserved.
//

import Foundation

protocol CharactersDataSource {
    
    func loadCharacters(page: Page, complete: ([Character]) -> Void, fail: () -> Void)
    
    func loadCharacter(characterId: String, complete: (Character) -> Void, fail: () -> Void)

    func saveCharacters(characters: Array<Character>, complete: () -> Void, fail: () -> Void)
}

File: common/src/main/kotlin/digital/wup/superheroapp/common/datasource/CharactersDataSource.kt
package digital.wup.superheroapp.common.datasource

import digital.wup.superheroapp.common.characters.domain.model.Page
import digital.wup.superheroapp.common.characters.domain.model.Character

interface CharactersDataSource {
    fun loadCharacters(page: Page, complete: (List<Character>) -> Unit, fail: () -> Unit)

    fun loadCharacter(characterId: Int, complete: (Character?) -> Unit, fail: () -> Unit)

    fun saveCharacters(characters: List<Character>, complete: () -> Unit, fail: () -> Unit)
}